x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -98,9 +95,8 @@ public void authenticate(final String action, final TransportRequest request, fi
                     // try-catch so any failure here is wrapped by `withRequestProcessingFailure`, whereas `authenticate` failures are not
                     // we should _not_ wrap `authenticate` failures since this produces duplicate audit events
                     try {
-                        final CrossClusterAccessSubjectInfo crossClusterAccessSubjectInfo = crossClusterAccessHeaders.subjectInfo();
-                        validate(crossClusterAccessSubjectInfo);
-                        writeAuthToContext(authcContext, authentication.toCrossClusterAccess(crossClusterAccessSubjectInfo), listener);
+                        final CrossClusterAccessSubjectInfo subjectInfo = crossClusterAccessHeaders.subjectInfo().cleanWithValidation();
```

Comment:
I wonder whether `crossClusterAccessHeaders` should not expose the subjectInfo that has not been cleaned and validated at all? That is, can we have something like `crossClusterAccessHeaders.getCleanAndValidatedSubjectInfo()` which internally calls `subjectInfo.cleanWithValidation()` and remove `crossClusterAccessHeaders.subjectInfo()`?